sway-core/src/semantic_analysis/ast_node/declaration/impl_trait.rs

```diff
@@ -482,13 +494,19 @@ fn type_check_trait_implementation(
     let mut functions_buf: Vec<ty::TyFunctionDeclaration> = vec![];
     let mut processed_fns = std::collections::HashSet::<Ident>::new();
 
+    let mut trait_fns = vec![];
+    for decl_id in trait_interface_surface.iter() {
+        match de_get_trait_fn(decl_id.clone(), block_span) {
+            Ok(decl) => trait_fns.push(decl),
+            Err(err) => errors.push(err),
+        }
+    }
```

Comment:
Seeing this pattern repeated a lot in this change... feels like it could be generalised.  Below with the `methods` and `de_get_function` too.  But I'm not sure what the best way would be since they're just slices.  Something like a `scan` or maybe just a fold, called `gather_decls` or something... but meh, it's not going to improve readability that much.  :shrug: 

If you want to `for` loop over a collection to get references then you can use `for elem in &elems {}` rather than calling `elems.iter()` usually, but in this case it's a slice so I think you can just remove the `.iter()`.